components/org.wso2.carbon.identity.fraud.detection.sift/src/main/java/org/wso2/carbon/identity/fraud/detection/sift/util/Util.java

```diff
@@ -0,0 +1,284 @@
+                String ip = (String) passedCustomParams.get(Constants.IP_KEY);
+                if (StringUtils.isNotBlank(ip)) {
+                    payload.put(Constants.IP_KEY, ip);
+                } else {
+                    payload.remove(Constants.IP_KEY);
+                }
+                passedCustomParams.remove(Constants.IP_KEY);
+            }
+
+            if (passedCustomParams.containsKey(Constants.SESSION_ID_KEY)) {
+                String sessionId = (String) passedCustomParams.get(Constants.SESSION_ID_KEY);
+                if (StringUtils.isNotBlank(sessionId)) {
+                    payload.put(Constants.SESSION_ID_KEY, sessionId);
+                } else {
+                    payload.remove(Constants.SESSION_ID_KEY);
+                }
+                passedCustomParams.remove(Constants.SESSION_ID_KEY);
+            }
+
+            if (passedCustomParams.containsKey(Constants.USER_AGENT_KEY)) {
+                String userAgent = (String) passedCustomParams.get(Constants.USER_AGENT_KEY);
+                if (StringUtils.isNotBlank(userAgent)) {
+                    Map<String, String> browserProperties = new HashMap<>();
+                    browserProperties.put(Constants.USER_AGENT_KEY, userAgent);
+                    payload.put(Constants.BROWSER_KEY, browserProperties);
+                } else {
+                    payload.remove(Constants.BROWSER_KEY);
+                }
+                passedCustomParams.remove(Constants.USER_AGENT_KEY);
+            }
+
+            // As the user_id is a mandatory field, it shouldn't be removed from the payload.
+            if (passedCustomParams.containsKey(Constants.USER_ID_KEY)) {
+                String userId = (String) passedCustomParams.get(Constants.USER_ID_KEY);
+                if (StringUtils.isNotBlank(userId)) {
+                    payload.put(Constants.USER_ID_KEY, userId);
+                }
+                passedCustomParams.remove(Constants.USER_ID_KEY);
+            }
+        }
+    }
+
+    public static Map<String, Object> getPassedCustomParams(Object[] paramMap) {
+
+        Map<String, Object> passedCustomParams = null;
+        if (paramMap.length == 1) {
+            if (paramMap[0] instanceof Map) {
+                passedCustomParams = (Map<String, Object>) paramMap[0];
+            } else {
+                throw new IllegalArgumentException("Invalid argument type. Expected paramMap " +
+                        "(Map<String, Object>).");
+            }
+        }
+        return passedCustomParams;
+    }
+
+    private static String getSiftApiKey(String tenantDomain) throws FrameworkException {
+
+        String apiKey = getSiftConfigs(tenantDomain).get(SIFT_API_KEY_PROP);
+        if (apiKey == null) {
+            throw new FrameworkException("Sift API key not found for tenant: " + tenantDomain);
+        }
+        return apiKey;
+    }
+
+    static Map<String, String> getSiftConfigs(String tenantDomain) throws FrameworkException {
```

Comment:
was using the default access modifier and making this package private intentional?